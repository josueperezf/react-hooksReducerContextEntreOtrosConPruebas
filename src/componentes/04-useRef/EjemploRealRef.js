import React, { useState } from 'react'
import '../02-useEffect/effect.css';
import {MultipleCustomHooks} from '../03-ejemplos/MultipleCustomHooks';

export const EjemploRealRef = () => {

    const [show, setshow] = useState(false);

    return (
        <div>
            <h1> Ejemplo Real de uso del hook Ref </h1>
            <hr/>
            { (show) && <MultipleCustomHooks/> }
            <button
                className="btn btn-primary mt-5"
                onClick={ ()=>{setshow( !show )}} >
                Show/Hide
            </button>
        </div>
    )
}
