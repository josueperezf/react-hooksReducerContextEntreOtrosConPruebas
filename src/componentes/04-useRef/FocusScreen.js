import React, { useRef } from 'react'
import '../02-useEffect/effect.css';
export const FocusScreen = () => {
    // la siguiente linea es para crear una referencia a un elemento html, en la propiedad current tiene lo que tenga nuestro element html
    const inputRef = useRef();
    const handleClick = ()=>{
        // console.log(inputRef);
        // document.querySelector('input').select();
        // current es como la propiedad que tiene el emento al que hacemos referencia
        inputRef.current.select();
    }
    return (
        <div>
            <h1>Focus screen</h1>
            <hr/>
            <input
                ref={inputRef}
                className="form-control"
                placeholder="Su nombre"
            />
            <button
                className="btn btn-outline-primary mt-5"
                onClick={ handleClick}
            >
                Focus
            </button>
        </div>
    )
}
