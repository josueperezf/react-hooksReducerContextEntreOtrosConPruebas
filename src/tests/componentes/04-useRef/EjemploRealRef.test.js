import { shallow } from "enzyme";
import { EjemploRealRef } from "../../../componentes/04-useRef/EjemploRealRef";

describe('Pruebas en  <EjemploRealRef/>',()=>{

    const wrapper = shallow(<EjemploRealRef/>);
    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
    });

    test('debe de mostrar el componente <MultipleCustomHooks/>', () => {
        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    });
    
    
});