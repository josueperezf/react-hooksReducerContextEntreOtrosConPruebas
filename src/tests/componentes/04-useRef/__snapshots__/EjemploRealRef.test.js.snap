// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en  <EjemploRealRef/> Debe mostrarse correctamente 1`] = `
<div>
  <h1>
     Ejemplo Real de uso del hook Ref 
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
